### ISSUE TITLE
Unable to pass "type: List" variables from data source to modules

### ISSUE THREAD

**Issue by guessi** [hashicorp/terraform#17837]
### Terraform Version

```
Terraform v0.11.7
+ provider.aws v1.14.0
+ provider.null v1.0.0
```

### Terraform Configuration Files

Please refer to the following demo code
- https://github.com/guessi/terraform-issue-demo

### Debug Output
```
<skipped>

2018/04/11 19:10:55 [TRACE] Graph after step *terraform.ReferenceTransformer:

module.dynamodb-table-test.aws_cloudwatch_metric_alarm.cloudwatch_alarm_read - *terraform.NodeValidatableResourceInstance
2018/04/11 19:10:55 [ERROR] root.dynamodb-table-test: eval: *terraform.EvalValidateResource, err: Warnings: []. Errors: [alarm_actions: should be a list]
2018/04/11 19:10:55 [ERROR] root.dynamodb-table-test: eval: *terraform.EvalSequence, err: Warnings: []. Errors: [alarm_actions: should be a list]

2018/04/11 19:10:55 [DEBUG] plugin: waiting for all plugin processes to complete...
Error: module.dynamodb-table-test.aws_cloudwatch_metric_alarm.cloudwatch_alarm_read: alarm_actions: should be a list
```

### Crash Output
N/A

### Expected Behavior
Should have no parsing error

### Actual Behavior
It will emit the following error message
```
Error: module.dynamodb-table-test.aws_cloudwatch_metric_alarm.cloudwatch_alarm_read: alarm_actions: should be a list
```

### Steps to Reproduce
```
$ git clone https://github.com/guessi/terraform-issue-demo.git
$ cd terraform-issue-demo
$ terraform init
$ terraform plan
```

### Additional Context
N/A

### References
N/A

**Comment by apparentlymart** [comment]
Hi @guessi! Sorry for this weird behavior, and for the delayed response.

What you saw here has the same root cause as #12263, which is now closed because a fix for it is in the master branch ready to be included in the forthcoming v0.12.0 release. Therefore I'm going to close this out. Thanks for reporting this, and sorry again for the late reply.

**Comment by guessi** [comment]
@apparentlymart it always have a workaround,
thanks for your effort, looking forward for releasing v0.12.0 :+1:
